draft-ietf-webpush-protocol.xml

```diff
@@ -1084,7 +1084,7 @@ HEADERS      [stream 82] +END_STREAM
             A push service MAY expire a subscription set at any time which MUST also
             expire all push message subscriptions in the set. If a user agent has an
             outstanding request to a push subscription set (<xref target="monitor-set"/>)
-            this can be signaled by returning a 400-series status code, such as 410 (Gone).
+            this can be signaled by returning a 410 (Gone) status code.

```

Comment:
While my wife reminds me I'm on vacation, might i suggest that 504 is a bit
more correct for Push Service gives up on delivery before TTL? It's a
server error, not a client one (thus the 500 class) and it's a timeout.

On Wed, Jun 1, 2016 at 12:45 PM, Brian Raymor <notifications@github.com>
wrote:

> In draft-ietf-webpush-protocol.xml
> <https://github.com/webpush-wg/webpush-protocol/pull/92#discussion_r65428972>
> :
>
> > @@ -1084,7 +1084,7 @@ HEADERS      [stream 82] +END_STREAM
> >              A push service MAY expire a subscription set at any time which MUST also
> >              expire all push message subscriptions in the set. If a user agent has an
> >              outstanding request to a push subscription set (<xref target="monitor-set"/>)
> > -            this can be signaled by returning a 400-series status code, such as 410 (Gone).
> > +            this can be signaled by returning a 410 (Gone) status code.
>
> Again, these are different failures. 404 is "push services gives up on
> delivery before TTL" and 410 is for expirations. The subscription set
> language should match the subscription language for expiration.
>
> It's unclear whether you're proposing to simply return 410 for both cases
> - but that's not reflected in this pull request.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/webpush-wg/webpush-protocol/pull/92/files/9749574d2c557e20828944384dde0ba510b5bb8a#r65428972>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AACLq8-1yBvTkY2J1pdsYEtPzqmWY4tWks5qHeFKgaJpZM4IrI5F>
> .
>